tests/unit/agents/common/test_agent.py

```diff
@@ -118,6 +124,56 @@ def test_create_chain(self, mock_models):
         # check step 2: model
         assert isinstance(chain.steps[1], RunnableLambda)
 
+    @pytest.mark.asyncio
+    @pytest.mark.parametrize(
+        "given_state, expected_inputs",
+        [
+            # Test case when agent_messages is empty, should use from messages field.
+            (
+                TestAgentState(
+                    is_last_step=False,
+                    messages=[HumanMessage(content="What is K8s?")],
+                    agent_messages=[],
+                    subtasks=[],
+                    k8s_client=Mock(spec=IK8sClient),
+                    my_task=SubTask(description="test", assigned_to="KubernetesAgent"),
+                ),
+                {
+                    "agent_messages": [HumanMessage(content="What is K8s?")],
+                    "query": "test",
+                },
+            ),
+            # Test case when agent_messages is non-empty, should not use from messages field.
+            (
+                TestAgentState(
+                    is_last_step=False,
+                    messages=[HumanMessage(content="What is K8s?")],
+                    agent_messages=[HumanMessage(content="What is an agent?")],
+                    subtasks=[],
+                    k8s_client=Mock(spec=IK8sClient),
+                    my_task=SubTask(description="test", assigned_to="KubernetesAgent"),
+                ),
+                {
+                    "agent_messages": [HumanMessage(content="What is an agent?")],
+                    "query": "test",
+                },
+            ),
+        ],
+    )
+    async def test_invoke_chain(
```

Comment:
this is not specific to the summarization topic, right? Why was it not here, before? Did you add it to see if invoking the chain is effected by the token limit feature?

Reply:
Wanted to test that the chain is invoked with input: `agent_messages`, not `messages`.